polkadot/runtime/common/src/slots/mod.rs

```diff
@@ -310,27 +384,112 @@ impl<T: Config> Pallet<T> {
 	// Useful when trying to clean up a parachain leases, as this would tell
 	// you all the balances you need to unreserve.
 	fn all_deposits_held(para: ParaId) -> Vec<(T::AccountId, BalanceOf<T>)> {
-		let mut tracker = sp_std::collections::btree_map::BTreeMap::new();
-		Leases::<T>::get(para).into_iter().for_each(|lease| match lease {
-			Some((who, amount)) => match tracker.get(&who) {
-				Some(prev_amount) =>
-					if amount > *prev_amount {
-						tracker.insert(who, amount);
+		ReservedAmounts::<T>::iter_prefix(para).collect::<Vec<_>>()
+	}
+
+	/// Returns how much deposit should be taken from the leaser for the current lease period of the
+	/// parachain. This is the maximum of all upcoming leases for the parachain with the same
+	/// leaser.
+	fn required_deposit(para: ParaId, leaser: &T::AccountId) -> BalanceOf<T> {
+		Leases::<T>::get(para)
+			.into_iter()
+			.map(|lease| match lease {
+				Some((who, amount)) =>
+					if &who == leaser {
+						amount
+					} else {
+						Zero::zero()
 					},
-				None => {
-					tracker.insert(who, amount);
-				},
-			},
-			None => {},
+				None => Zero::zero(),
+			})
+			.max()
+			.unwrap_or_else(Zero::zero)
+	}
+
+	/// Returns true if the current lease period is ending within the next `T::EarliestRefundPeriod`
+	/// blocks.
+	fn lease_period_ending_soon(now: BlockNumberFor<T>) -> bool {
```

Comment:
"Soon" is very ambiguous.
```suggestion
	fn within_early_refund_period(now: BlockNumberFor<T>) -> bool {
```